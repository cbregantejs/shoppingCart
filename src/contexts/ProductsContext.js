import React, { createContext } from 'react';
import { useFechProducts } from '../hooks/useFechProducts';
export const ProductsContext = createContext()

const ProductsContextProvider = ({children}) => {

    const { data: products } = useFechProducts();

    return ( 
        <ProductsContext.Provider value={{products}} >
            { children }
        </ProductsContext.Provider>
     );
}
 
export default ProductsContextProvider;