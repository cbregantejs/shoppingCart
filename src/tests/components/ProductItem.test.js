import * as React from 'react';
import { shallow } from 'enzyme';
import {ProductItem} from '../../pages/store/ProductItem';
import { CartContext } from '../../contexts/CartContext';

describe('Pruebas en <ProductItem />', () => {

  const product = {
    "id": 1,
    "title": "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops",
    "price": 109.95,
    "description": "Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday",
    "category": "men's clothing",
    "image": "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
  }
  const wrapper = shallow( 
    <CartContext.Provider>
      <ProductItem  product={product} />
    </CartContext.Provider>
  )

  test('debe mostrar el producto correctamente', () => {
    expect ( wrapper ).toMatchSnapshot();
  })

})
