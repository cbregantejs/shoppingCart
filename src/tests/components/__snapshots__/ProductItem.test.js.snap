// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ProductItem /> debe mostrar el producto correctamente 1`] = `
<ProductItem
  product={
    Object {
      "category": "men's clothing",
      "description": "Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday",
      "id": 1,
      "image": "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
      "price": 109.95,
      "title": "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops",
    }
  }
/>
`;
